modules/rates-resolver/src/main/scala/fi/spectrumlabs/rates/resolver/services/MetadataService.scala

```diff
@@ -0,0 +1,36 @@
+package fi.spectrumlabs.rates.resolver.services
+
+import cats.syntax.eq._
+import cats.syntax.parallel._
+import cats.{Functor, Monad, Parallel}
+import fi.spectrumlabs.core.models.domain.AssetClass
+import fi.spectrumlabs.rates.resolver.{AdaAssetClass, AdaDecimal, AdaMetadata}
+import fi.spectrumlabs.rates.resolver.gateways.Metadata
+import fi.spectrumlabs.rates.resolver.models.{Metadata => Meta}
+import tofu.logging.Logs
+import tofu.syntax.monadic._
+
+trait MetadataService[F[_]] {
+  def getTokensInfo(tokens: List[AssetClass]): F[List[Meta]]
```

Comment:
Looks like this method should be in one module with getTokenInfo. Models are the same, so Metadata and MetadataServer distinction doesn't make sense. 
You can make Metadata a pure client whose methods return models as-is without any conversions. Then do all the conversions in the MetadataService.

Reply:
I wanna do it like this:
Client is `trait Meta[F[_]] { def getAssetMeta(asset: AssetClass): F[Response] }` and service is `trait MetaService[F[_]] { def getMetaForAssets(in: List[AssetClass]): F[List[AssetMeta]] }`